dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,677 @@
+                        Matchers.hasToString(url + "/bitstreams/" + bitstream1.getID() + "/download")))
+                .andExpect(jsonPath("$.linkset[0].item[0].type",
+                        Matchers.hasToString(bitstream1MimeType)))
+                .andExpect(jsonPath("$.linkset[0].item[1].href",
+                        Matchers.hasToString(url + "/bitstreams/" + bitstream2.getID() + "/download")))
+                .andExpect(jsonPath("$.linkset[0].item[1].type",
+                        Matchers.hasToString(bitstream2MimeType)))
+                .andExpect(jsonPath("$.linkset[0].anchor",
+                        Matchers.hasToString(url + "/entities/publication/" + item.getID())));
+    }
+
+    @Test
+    public void findOneItemThatIsInWorkspaceJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        WorkspaceItem workspaceItem = WorkspaceItemBuilder.createWorkspaceItem(context, collection)
+                .withTitle("Workspace Item")
+                .build();
+        itemService.addMetadata(context, workspaceItem.getItem(), "dc", "identifier", "doi", Item.ANY, doi);
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + workspaceItem.getItem().getID() + "/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))));
+    }
+
+    @Test
+    public void findOneWithdrawnItemJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Withdrawn Item")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .withdrawn()
+                .build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))));
+    }
+
+    @Test
+    public void findOneEmbargoItemJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Withdrawn Item")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .withIssueDate("2017-11-18")
+                .withEmbargoPeriod("2 week")
+                .build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json"))
+                .andExpect(status().isOk())
```

Comment:
An embargoed Item should NOT be possible to find in the Signposting links.  This should return a 404.

Reply:
Fixed